feat(doc): document Lua alternative for vim.fn (#33313)
neovim/neovim: runtime/doc/vimfn.txt

                  (`0|1`)

bufadd({name})                                                        *bufadd()*
		Lua: Prefer |nvim_create_buf()| with |nvim_buf_set_name()|.

> 

Reviewer: Really?

Reviewer 2: some of these are more like "see also" rather than "preferred", but maybe we can adjust in a followup.

                  (`integer`)

> environ()                                                            *environ()*
> 		Lua: Prefer |uv.os_environ()| for a real environment table;
> 		|vim.env| for keyed access.

Reviewer 2: nit/nonblocker, what is meant by "real", could just drop that or does it mean "full"? :)
```suggestion
environ()                                                            *environ()*
		Lua: Prefer |uv.os_environ()| for the full environment table;
		|vim.env| for keyed access.
```

                  (`integer`)

environ()                                                            *environ()*

> 		Lua: Prefer |vim.env|.

Reviewer: I don't think so...

Author: You make it seem like it obvious. Could you elaborate just a little.

Reviewer: `vim.env` is a metatable so you cannot iterate over its keys. And `uv.os_environ()` will be a exact alternative anyway.

                  (`integer`)

mode([{expr}])                                                          *mode()*
		Lua: Prefer |nvim_get_mode()|; use the `.mode` field for the

> 		mode string, and note that full-mode behavior differs.

Reviewer: "full-mode behavior differs" is a bit confusing.

                  (`integer`)

repeat({expr}, {count})                                               *repeat()*

> 		Lua: Prefer |string.rep()| for strings; lists and blobs need

Reviewer: Aren't strings and blobs the same in Lua?

                  (`string`)

winheight({nr})                                                    *winheight()*
		Lua: Prefer |nvim_win_get_height()| for valid windows; invalid

> 		IDs error instead of returning -1.

Reviewer: Multigrid and winbar behavior differs. `winheight()` gets the inner height. `nvim_win_get_height()` gets the outer height (e.g. it includes winbar unlike `winheight()`).

                  (`vim.fn.winsaveview.ret`)

winwidth({nr})                                                      *winwidth()*
		Lua: Prefer |nvim_win_get_width()| for valid windows; invalid

> 		IDs error instead of returning -1.

Reviewer: Multigrid behavior differs. `winwidth()` gets the inner width. `nvim_win_get_width()` gets the outer width.
